### DISCUSSION TITLE
If you have drag-n-drop enabled are you required to test it that way?

### DISCUSSION THREAD

**Discussion by HamboneWilson** [abramenal/cypress-file-upload#287] [Q&A]
Is { subjectType: 'drag-n-drop' } required to test html5 upload elements with drag-n-drop enabled? Or can you test with or without it to emulate either pressing the upload button or dropping the file in?

**Comment by abramenal** [comment]
Subject type is needed to determine events that are expected from this component: https://github.com/abramenal/cypress-file-upload/blob/1ae2d71821780c2f5f720c2e3a6be0593fa70825/src/constants.js#L12
It may also alter a bit the way fixtures are attached to that specific element.

I believe ability to emulate click-and-choose-file behaviour is determined by every drag-n-drop component, but it is indeed a good idea to try this out. Would be really curious to hear someone's feedback whether it is actually working that way
